src/Parallel/CharmMain.cpp

```diff
@@ -19,6 +19,23 @@ namespace charmxx {
 template <class...>
 using void_t = void;
 
+template <class T>
```

Comment:
This appears to be only slightly different from just writing `T::value_type` (as implied by the name).  Could you add a comment explaining why it is needed?

Reply:
It's a different type, with a non-const Key, I don't think that needs explanation